### PR TITLE
refactor: Static File Handler Overhaul

### PR DESCRIPTION
**Pull request by nielsenko**
## Description
Static File Handler Overhaul

This PR reimplements the static file serving functionality to provide better HTTP compliance, improved performance, and enhanced security.

### Key Improvements

**HTTP Compliance & Standards**
- Full support for conditional requests (`If-None-Match`, `If-Modified-Since`)
- Complete HTTP range request implementation including multipart ranges
- Proper ETag generation using SHA-1 content hashing
- Correct `Last-Modified`, `Accept-Ranges`, and `Cache-Control` headers
- Proper HTTP method validation (405 responses for unsupported methods)

**Performance Enhancements**
- LRU cache for file metadata to reduce filesystem operations
- Efficient streaming with proper content-length handling
- Smart MIME type detection using both file extensions and magic numbers

**Security & Reliability**
- Removed directory listing functionality (security concern)
- Enhanced symbolic link validation to prevent directory traversal
- Better path sanitization and validation
- Proper handling of edge cases and malformed requests

### API Changes

**Simplified Configuration**
- Removed complex configuration options: `serveFilesOutsidePath`, `defaultDocument`, `listDirectories`, `useHeaderBytesForContentType`
- Added streamlined options: `defaultResponse`, `mimeResolver`, `cacheControl`
- New `createFileHandler()` function for serving single files

**Smart Content-Type Handling**
- Default encoding is now `null` for binary files, `utf8` only for text files
- Default MIME type changed from `text/plain` to `application/octet-stream`
- Added `MimeType.isText` getter for better content type logic

**Enhanced Routing Context**
- Added `matchedPath` and `remainingPath` properties to routing middleware
- Better support for nested routing scenarios

### Testing

- Comprehensive test suite covering all HTTP scenarios
- Tests for conditional requests, range requests, edge cases, and security
- Removed tests for deprecated features (directory listings, default documents)

### Dependencies

- Added `crypto` package for ETag generation
- Updated dev dependencies to latest versions

This overhaul brings Relic's static file serving up to production standards with proper HTTP semantics, making it suitable for serving static assets in web applications.

## Related Issues
- Closes: #8 
- Fixes: #116

## Pre-Launch Checklist
Please ensure that your PR meets the following requirements before submitting:

- [x] This update focuses on a single feature or bug fix. (For multiple fixes, please submit separate PRs.)
- [x] I have read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code using [dart format](https://dart.dev/tools/dart-format).
- [x] I have referenced at least one issue this PR fixes or is related to.
- [x] I have updated/added relevant documentation (doc comments with `///`), ensuring consistency with existing project documentation. 
- [x] I have added new tests to verify the changes.
- [x] All existing and new tests pass successfully.
- [x] I have documented any breaking changes below.

## Breaking Changes
- Directory listing functionality removed entirely
- `createStaticHandler()` parameters simplified - some options no longer available
- Default MIME type and encoding behavior changed
- `createFileHandler()` API simplified

## Summary by CodeRabbit

* **New Features**
  * Static file server: ETag/If-None-Match, If-Modified-Since, single/multipart Range support, configurable Cache-Control, improved MIME detection, 405 for unsupported methods (GET/HEAD allowed).

* **Refactor**
  * Static handler rewritten with streaming, per-file metadata and stricter symlink/path behavior; default document is now served via a configurable default handler; body defaults favor binary (application/octet-stream) unless text mime detected; routing exposes matched and remaining path info.

* **Bug Fixes**
  * Safer path handling; pathParameters returns empty map when no route matches.

* **Chores**
  * Dependency updates (crypto, config, dev tooling).

* **Tests**
  * Extensive tests added/updated for ranges, conditional requests, content types, cache-control, not-found and unsupported methods; directory-listing tests removed.

### CODE REVIEW

**Comment by nielsenko** [comment]
@coderabbitai review

File: lib/src/io/static/static_handler.dart [hunk 1]
```diff
@@ -1,327 +1,443 @@
...
-    start = startMatch;
-    end = endMatch ?? actualLength - 1;
+    start = fileSize - (range.end ?? 0);
+    end = fileSize;
   }
+  // Ensure start is within bounds
+  start = start.clamp(0, fileSize);
+  end = end.clamp(start, fileSize);
+  return (start, end);
+}
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply lib/src/io/static/static_handler.dart:392]: So we agree this is correct. Start and end are the seek positions in the file we are about to read.

Not sure where you got diff regarding `startMatch` from? Are you inventing stuff?

File: lib/src/io/static/static_handler.dart [hunk 2]
```diff
@@ -1,327 +1,443 @@
...
+  final String boundary,
+  final int start,
+  final int end,
+) async {
+  int totalBytes = 0;
+
+  // Write part header
+  final partHeader = '\r\n--$boundary\r\n'
+      'Content-Type: ${fileInfo.mimeType!.toHeaderValue()}\r\n'
+      'Content-Range: bytes $start-${end - 1}/${fileInfo.stat.size}\r\n\r\n';
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply lib/src/io/static/static_handler.dart:408]: Good catch, will go about it a bit differently.

File: test/static/if_range_test.dart [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      'Given an If-Range header with a matching Last-Modified date, '
+      'when a range request is made, '
+      'then a 206 Partial Content response with partial content is returned',
+      () async {
+    final rootPath = p.join(d.sandbox, 'test_file.txt');
+    final modified = File(rootPath).statSync().modified.toUtc();
+    final headers = Headers.build((final mh) => mh
+      ..range = RangeHeader(ranges: [Range(start: 0, end: 4)])
+      ..ifRange = IfRangeHeader(lastModified: modified));
+
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply test/static/if_range_test.dart:65]: Fixed

File: test/static/if_modified_since_test.dart [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    final headers =
+        Headers.build((final mh) => mh.ifModifiedSince = lastModified);
+
+    final response =
+        await makeRequest(handler, '/test_file.txt', headers: headers);
+
+    expect(response.statusCode, HttpStatus.notModified);
+    expect(response.body.contentLength, 0);
+    expect(await response.readAsString(), isEmpty);
+  });
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply test/static/if_modified_since_test.dart:35]: Fixed

**Comment by nielsenko** [comment]
@coderabbitai review

File: lib/src/io/static/static_handler.dart [hunk 5]
```diff
@@ -1,327 +1,445 @@
...
+/// The [mimeResolver] can be provided to customize MIME type detection.
+/// The [cacheControl] header can be customized; defaults to no-cache with private cache.
+Handler createFileHandler(
+  final String filePath, {
+  final MimeTypeResolver? mimeResolver,
+  final CacheControlHeader? cacheControl,
+}) {
+  final file = File(filePath);
+  if (!file.existsSync()) {
+    throw ArgumentError(
```
Comment (SandPod) [inline lib/src/io/static/static_handler.dart:99]: Comment: Nice that we fail early on this 🌟

File: lib/src/io/static/static_handler.dart [hunk 6]
```diff
@@ -1,327 +1,445 @@
...
-        final uri = request.requestedUri;
-        if (!uri.path.endsWith('/')) return _redirectToAddTrailingSlash(uri);
-        return listDirectory(fileSystemPath, fsPath);
-      }
-    }
+  return (final NewContext ctx) async {
+    return await _serveFile(
+      file,
+      mimeResolver ?? _defaultMimeTypeResolver,
+      cacheControl ?? CacheControlHeader(noCache: true, privateCache: true),
```
Comment (SandPod) [inline lib/src/io/static/static_handler.dart:107]: Question(NIT): Any reason for `CacheControlHeader` not to be const?

File: lib/src/io/static/static_handler.dart [hunk 7]
```diff
@@ -1,327 +1,445 @@
...
+/// Serves a file with full HTTP semantics including conditional requests and ranges.
+Future<ResponseContext> _serveFile(
+  final File file,
+  final MimeTypeResolver mimeResolver,
+  final CacheControlHeader? cacheControl,
+  final NewContext ctx,
+) async {
+  // Validate HTTP method
+  final method = ctx.request.method;
+  if (!_isMethodAllowed(method)) return _methodNotAllowedResponse(ctx);
```
Comment (SandPod) [inline lib/src/io/static/static_handler.dart:122]: Suggestion: Add documentation for this behavior in the public api (`reateFileHandler` and `createStaticHandler`).

Same for the other possible responses and functionality, it will help developers understand what we have automated.

File: lib/src/io/static/static_handler.dart [hunk 8]
```diff
@@ -1,327 +1,445 @@
...
+  // Check if cache is valid
+  if (cachedInfo != null &&
+      stat.size == cachedInfo.stat.size &&
+      !stat.changed.isAfter(cachedInfo.stat.changed)) {
+    return cachedInfo;
+  }
 
-        await file.openRead(0, length).listen(byteSink.add).asFuture<void>();
+  // Generate new file info
+  final etag = await _generateETag(file);
```
Comment (SandPod) [inline lib/src/io/static/static_handler.dart:174]: Comment: This will provide a poor mans cache busting if the `revalidate` header is used, could be useful until we have built a proper cache busting mechanism in Serverpod.

In serverpod, we could use this pattern until we have proper cache-busting built.

Comment (SandPod) [reply lib/src/io/static/static_handler.dart:174]: Btw, Is this also a differentiator between us and shelf?

File: lib/src/io/static/static_handler.dart [hunk 9]
```diff
@@ -1,327 +1,445 @@
...
+  final NewContext ctx,
   final File file,
+  final _FileInfo fileInfo,
   final Headers headers,
-) {
-  final range = request.headers.range;
-  if (range == null) return null;
+  final List<Range> ranges,
+) async {
+  final boundary = 'RelicMultipartBoundary-${Random().nextInt(1000000)}';
```
Comment (SandPod) [inline lib/src/io/static/static_handler.dart:334]: Question: This will tag all multipart request from relic with this. Do we want to be able to blend into the crowd here?
Is there a general boundary that can be used?

File: test/static/basic_file_test.dart [hunk 10]
```diff
@@ -215,17 +215,20 @@ void main() {
...
-      expect(response.mimeType, isNull);
+      expect(response.mimeType, MimeType.octetStream);
     });
 
     test('when accessing header_bytes_test_image then it should be image/png',
         () async {
-      final handler =
-          createStaticHandler(d.sandbox, useHeaderBytesForContentType: true);
+      final handler = createStaticHandler(
+        d.sandbox, /* useHeaderBytesForContentType: true */
```
Comment (SandPod) [inline test/static/basic_file_test.dart:230]: Suggestion: Remove commented out comments.

Comment (SandPod) [reply test/static/basic_file_test.dart:230]: Is this important for the test?

File: test/static/cache_control_test.dart [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+      final response = await makeRequest(handler, '/test_file.txt');
+
+      expect(response.statusCode, HttpStatus.ok);
+      expect(response.headers.cacheControl, isNotNull);
+      expect(response.headers.cacheControl!.maxAge, 3600);
+      expect(response.headers.cacheControl!.publicCache, isTrue);
+      expect(response.headers.cacheControl!.mustRevalidate, isTrue);
+      expect(response.headers.cacheControl!.noCache, isFalse);
+      expect(response.headers.cacheControl!.noStore, isFalse);
```
Comment (SandPod) [inline test/static/cache_control_test.dart:41]: Question: Are these relevant for the test? If not, please remove, else please motivate or break out into a separate test.

File: lib/src/io/static/static_handler.dart [hunk 12]
```diff
@@ -1,327 +1,445 @@
...
 ///
-/// If [useHeaderBytesForContentType] is `true`, the contents of the
-/// file will be used along with the file path to determine the content type.
+/// If the requested path doesn't correspond to a file, returns [defaultResponse].
+/// Directory listings are not supported for security reasons.
 ///
-/// Specify a custom [contentTypeResolver] to customize automatic content type
-/// detection.
+/// The [mimeResolver] can be provided to customize MIME type detection.
+/// The [cacheControl] header can be customized; defaults to no-cache with private cache.
```
[the remark this answers is not in the document]
Comment (SandPod) [reply lib/src/io/static/static_handler.dart:41]: Question(For my knowledge): Where do we define this value?

File: test/static/cache_control_test.dart [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+    });
+
+    test(
+        'Given no Cache-Control header is specified on the server, '
+        'when a file is served, '
+        'then the response includes no Cache-Control header', () async {
+      final response = await makeRequest(handler, '/test_file.txt');
+
+      expect(response.statusCode, HttpStatus.ok);
+      expect(response.headers.cacheControl, isNull);
```
Comment (SandPod) [inline test/static/cache_control_test.dart:51]: Suggestion: According to the documentation, this should be `noCache` and `private`. That doesn't seem to correspond to this.

File: test/static/cache_control_test.dart [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    });
+
+    test(
```
Comment (SandPod) [inline test/static/cache_control_test.dart:54]: Question: Isn't this a duplication of the first test?

File: test/static/default_document_test.dart [hunk 15]
```diff
@@ -16,24 +17,7 @@ void main() {
...
-
-      for (final val in invalidValues) {
-        expect(() => createStaticHandler(d.sandbox, defaultDocument: val),
-            throwsArgumentError);
-      }
-    });
-  });
-
-  group('Given no default document specified', () {
+  group('Given no default handler specified', () {
```
Comment (SandPod) [inline test/static/default_document_test.dart:20]: Suggestion: Clarify the description of these tests: "when accessing a non existing directory" or "when accessing a non existing file". 

The paths requires detailed knowledge about the test setup to understand (that is not reflected in the test description).

File: test/static/default_document_test.dart [hunk 16]
```diff
@@ -66,11 +50,11 @@ void main() {
     });
   });
 
-  group('Given a default document specified', () {
+  group('Given a default handler specified', () {
```
Comment (SandPod) [inline test/static/default_document_test.dart:53]: Suggestion: Specify the behavior of the default handler in the Given part, otherwise the `then` part doesn't really make sense.

File: test/static/if_none_match_test.dart [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+void main() {
+  const fileContent = '0123456789ABCDEF'; // 16 bytes
+  late Handler handler;
+
+  setUp(() async {
+    await d.file('test_file.txt', fileContent).create();
+    handler = createStaticHandler(d.sandbox);
+  });
+
+  test(
```
Comment (SandPod) [inline test/static/if_none_match_test.dart:18]: Question: Shouldn't we have a test for etag with the revalidate header? Or does this result in a `If-None-Match` or `If-Modified-Since` request?

File: test/static/if_range_test.dart [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+    final headers = Headers.build((final mh) => mh
+      ..range = RangeHeader(ranges: [Range(start: 0, end: 4)])
+      ..ifRange = IfRangeHeader(etag: ETagHeader(value: etag)));
+
+    final response =
+        await makeRequest(handler, '/test_file.txt', headers: headers);
+
+    expect(response.statusCode, HttpStatus.partialContent);
+    expect(response.body.contentLength, 5);
+    expect(response.readAsString(), completion(fileContent.substring(0, 5)));
```
Comment (SandPod) [inline test/static/if_range_test.dart:35]: Question(For my knowledge): Is the Range in the header inclusive while substring is exclusive?

File: test/static/not_found_test.dart [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+  const fileContent = '0123456789ABCDEF'; // 16 bytes
+  late Handler handler;
+
+  setUp(() async {
+    await d.file('test_file.txt', fileContent).create();
+    await d.dir('test_directory').create();
+    handler = createStaticHandler(d.sandbox);
+  });
+
+  group('Given non-existent resources', () {
```
Comment (SandPod) [inline test/static/not_found_test.dart:19]: Question: aren't these tests duplicated with the tests in `default_document_test`? Perhaps we could remove the rainy day tests from there or combine them?

File: test/static/default_document_test.dart [hunk 20]
```diff
@@ -16,24 +17,7 @@ void main() {
     ]).create();
```
Comment (SandPod) [inline test/static/default_document_test.dart:17]: Suggestion: Rename this file to match the value it is configured through. `default_handler_test.dart`

File: test/static/not_found_test.dart [hunk 21]
```diff
@@ -0,0 +1,124 @@
...
+          await makeRequest(handler, '/%2E%2E%2F%2E%2E%2Fetc%2Fpasswd');
+
+      expect(response.statusCode, HttpStatus.notFound);
+      expect(
+          response.body.contentLength, greaterThan(0)); // "Not Found" message
+      expect(await response.readAsString(), contains('Not Found'));
+    });
+  });
+
+  group('Given symbolic links (if supported)', () {
```
Comment (SandPod) [inline test/static/not_found_test.dart:97]: Suggestion: Consolidate this with the tests in `symbolic_link_test.dart`, I think this scenario is duplicated now.

**Review by SandPod — CHANGES_REQUESTED** [review]
Big PR -> a lot of questions.

Overall, it looks good. This will be a solid addition. Would be nice to know what of this is not supported by shelf today.

I had a couple of questions that I wanted to discuss. Some things might be better discussed during a call.

There are also a couple of test fix suggestions.

File: lib/src/io/static/static_handler.dart (hunk 8 above)
Comment (SandPod) [inline lib/src/io/static/static_handler.dart:174]: Question: Should we run this in an isolate? And should this be run parallel to detectingMimeType?

**Comment by nielsenko** [comment]
> ... Would be nice to know what of this is not supported by shelf today. ...

- If-None-Modified request header. The server will hand out ..
- ETag response headers. We support strong SHA1 etags cached in an ..
- LRU cache for quick lookup of file metadata (`_FileInfo(this.mimeType, this.stat, this.etag)`).
- Multi-range `Range` headers. Shelf only support single-range request headers
- `If-Range` (both etag and lastModified),
- Proper verb validation (no PUT, POST, etc), with ..
- `Allow` response header.
- Remember to set `Accept-Ranges`, `Last-Modified`, `ETag` header on all replies, not just sometimes.
- Better error handling in general (in particular around 416).
- Never allow traversal outside root (whether by symbolic link, or parent traversal '..')
- Use typed headers.
- Higher test coverage (90%+).
- Maybe more 😄

File: test/static/not_found_test.dart (hunk 19 above)
Comment (nielsenko) [reply test/static/not_found_test.dart:19]: There are some duplication, yes. Will fix.

File: lib/src/io/static/static_handler.dart (hunk 8 above)
Comment (nielsenko) [reply lib/src/io/static/static_handler.dart:174]: We could for `_generateETag`, but I doubt parallelising with mimeType detection will give much as detection is much faster, and hence the particular request will still have to wait about the same time.

My thinking around performance and isolates has (sofar) been to run a request per isolate. (See _multi_isolate.dart_ example)

Comment (nielsenko) [reply lib/src/io/static/static_handler.dart:174]: Yes, shelf_static does not support ETags/If-None-Match.

I'll need a bit more context to comment on your cache-busting idea.

File: test/static/symbolic_link_test.dart [hunk 22]
```diff
@@ -33,129 +33,56 @@ void main() {
     Link(p.join(d.sandbox, 'alt_root', 'link_dir')).createSync(originalsDir);
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply test/static/symbolic_link_test.dart:33]: I'll look into moving the setup into the tests - to make it easier to reason about a single test case.

File: test/static/symbolic_link_test.dart [hunk 23]
```diff
@@ -33,129 +33,56 @@ void main() {
...
       expect(response.body.contentLength, 13);
       expect(response.readAsString(), completion('<html></html>'));
     });
+  });
 
-    group('Given links under root dir', () {
-      test(
-        'when accessing a sym linked file in a real dir then it returns the file content',
+  group('Given links not under root dir', () {
+    test('when accessing a sym linked file in a real dir then it returns a 404',
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply test/static/symbolic_link_test.dart:72]: Yes. I'm being strict here for security reasons. Shelf has a flag that relaxes this, that I decided to get rid of, but perhaps I'm being over eager. 

Should we allow symlinks out of the root, if explicitly requested?

File: lib/src/io/static/static_handler.dart [hunk 24]
```diff
@@ -1,327 +1,445 @@
...
-
-  final mimeResolver = contentTypeResolver ?? _defaultMimeTypeResolver;
+  return (final NewContext ctx) async {
+    final requestPath = ctx.remainingPath.path;
+    final filePath = p.join(resolvedRootPath, requestPath.substring(1));
 
-  return respondWith((final Request request) {
-    final segs = [fileSystemPath, ...request.url.pathSegments];
+    // Ensure file exists and is not a directory
+    final entityType = FileSystemEntity.typeSync(filePath, followLinks: false);
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply lib/src/io/static/static_handler.dart:62]: `followLinks: false` will just return that the entityType is a link, if that is the case, instead of trying to follow the links. 
This avoids an exception, if the path doesn't even exists.

The actual symlink resolve happens later where we also check bounds:
```dart
    final resolvedFilePath = file.resolveSymbolicLinksSync();
    if (!p.isWithin(resolvedRootPath, resolvedFilePath)) {
      return defaultHandler!(ctx);
    }
```

File: lib/src/io/static/static_handler.dart (hunk 6 above)
Comment (nielsenko) [reply lib/src/io/static/static_handler.dart:107]: Yes, it does some validation, regarding setting both `publicCache`, and `privateCache`, or both `maxAge` and `staleWhileRevalidate`, which is illegal - hence it cannot be const.

We could consider using an enum instead of the two nullable bool flags?

File: test/static/cache_control_test.dart (hunk 13 above)
Comment (nielsenko) [reply test/static/cache_control_test.dart:51]: Fixed

File: test/static/cache_control_test.dart (hunk 14 above)
Comment (nielsenko) [reply test/static/cache_control_test.dart:54]: I tend to agree - removed

File: test/static/symbolic_link_test.dart (hunk 22 above)
[the remark this answers is not in the document]
Comment (nielsenko) [reply test/static/symbolic_link_test.dart:29]: I have added better comments and descriptions, like:
```dart
    // Setup the following directory hierarchy:
    //
    // originals/
    //   index.html
    // alt_root/
    //   link_index.html ../originals/index.html
    //   link_dir -> ../originals/
    // link_index.html -> originals/index.html
    // link_dir -> originals/
```

File: test/static/default_document_test.dart (hunk 20 above)
Comment (nielsenko) [reply test/static/default_document_test.dart:17]: Done

File: test/static/not_found_test.dart (hunk 21 above)
Comment (nielsenko) [reply test/static/not_found_test.dart:97]: Done

File: test/static/not_found_test.dart (hunk 19 above)
Comment (nielsenko) [reply test/static/not_found_test.dart:19]: Done

File: lib/src/middleware/routing_middleware.dart [hunk 25]
```diff
@@ -32,10 +39,14 @@ class _RoutingMiddlewareBuilder<T> {
   Handler _meddle(final Handler next) {
     return (final ctx) async {
       final req = ctx.request;
-      final url = ctx.request.url; // TODO: Use requestUri
-      final match = _router.lookup(req.method.convert(), url.path);
+      final match =
```
[the remark this answers is not in the document]
Comment (SandPod) [reply lib/src/middleware/routing_middleware.dart:42]: I noticed your other PR as well, so we can take that discussion there. But this is not prioritized for now.

File: test/static/symbolic_link_test.dart (hunk 23 above)
[the remark this answers is not in the document]
Comment (SandPod) [reply test/static/symbolic_link_test.dart]: We could start being very restrictive and then add options to relax it if requested. It is possible to work around this rule by copying the content into the server. If you deploy this somewhere, you will likely not have a separate project with the assets.

Decision: Keep as it is.

File: lib/src/io/static/static_handler.dart (hunk 24 above)
[the remark this answers is not in the document]
Comment (SandPod) [reply lib/src/io/static/static_handler.dart:69]: Aah, got it! Thanks for the explanation!

File: lib/src/io/static/static_handler.dart (hunk 7 above)
Comment (SandPod) [reply lib/src/io/static/static_handler.dart:129]: Shouldn't this be reflected in `createFileHandler` as well?

File: lib/src/io/static/static_handler.dart (hunk 9 above)
Comment (SandPod) [reply lib/src/io/static/static_handler.dart:341]: @coderabbitai Can you create an issue for researching what is commonly used as a boundry and for chaning that to adhere to that standard. We are concerned that this boundry might be used to fingerprint the software running on the server.

**Review by SandPod — APPROVED** [review]
LGTM! 🚀
